java-checks/src/main/java/org/sonar/java/checks/MismatchPackageDirectoryCheck.java

```diff
@@ -47,7 +47,10 @@ public void visitCompilationUnit(CompilationUnitTree tree) {
       String dir = javaFile.getParent();
       if (!dir.endsWith(packageName)) {
         String dirWithoutDots = dir.replace(".", File.separator);
-        String issueMessage = MessageFormat.format(MESSAGE, javaFile.getName(), packageName, dir);
+        int srcIndex = dir.indexOf("src");
+        String truncatedPath = dir.substring(srcIndex == -1 ? 0 : srcIndex);
```

Comment:
I understand that by convention, java files should be placed under `src/{main|test}/java`, but users can customize the location where their sources are located. Can we use the module specific source directory to match?

Reply:
After discussion, a simpler approach is to use `sonar.projectBaseDir` as the base root